### DISCUSSION TITLE
[vue] keepPreviousData lost data when used from another route

### DISCUSSION THREAD

**Discussion by Ask1n** [TanStack/query#4795] [Q&A] [answered]
Hi, i use keepPreviousData and it works fine inside one route, but when switching to another, the data is lost. 

/user/:slug
/category/:slug
used one component with useQuery

/user/1 -> /user/2 work fine
/category/3 - /category/4 same

/user/1 -> /category/5 lost previous data

how can this be solved?

**Comment by TkDodo** [comment, marked as answer]
`keepPreviousData` works on a per-observer basis. Meaning if you have one call of `useQuery`, and you render the same component again and the key changes, it will keep the data it had before.

If you change the route and render a completely different component, you'll get a new observer, so there is not previous data to keep. You would need to re-render _the same component_ on a route change to achieve that.
